<?xml version="1.0"?>
<launch>
    <arg name="robot_name"      default="$(env ROBOT_HOST)"/>
    <arg name="master_name"      default="$(env ROBOT_MASTER)"/>
    <arg name="slam_methods"    default="gmapping" doc="slam type 
        [gmapping, cartographer, hector, karto]"/>

    <arg name="gmapping"        default="gmapping"/>
    <arg name="cartographer"    default="cartographer"/>
    <arg name="hector"          default="hector"/>
    <arg name="karto"           default="karto"/>
    <arg name="x"               default="0.0"/>
    <arg name="y"               default="1.0"/>

    <include file="$(find jetauto_slam)/launch/include/jetauto_robot.launch">
        <arg name="sim" value="false"/>
	<arg name="app" value="false"/>
        <arg name="robot_name" value="$(arg robot_name)"/>
	<arg name="master_name" value="$(arg master_name)"/>
    </include>

    <node name="set_odom" pkg="jetauto_multi" type="set_odom.py" output="screen">
        <param name="robot_name"    value="$(arg robot_name)"/>
        <param name="slam_methods"  value="$(arg slam_methods)"/>
        <param name="x"  value="$(arg x)"/>
        <param name="y"  value="$(arg y)"/>
    </node>
</launch>
